### ISSUE TITLE
message_loop starts DDoS attack

### ISSUE THREAD

**Issue by long-exposure** [nickoala/telepot#205]
Recently I started playing with Telepot (10.4) and it worked out of the box. I'm using the message_loop, where I have seen that Telepot sends getUpdates POST requests every 20 seconds. This is the good case when Telegram answer with a 200 HTTP response and everything is fine.

Now I have observed an outage on the Telegram side, i.e. they sent a 502 HTTP response on every getUpdates request. In this case Telepot immediately repeats sending of the getUpdates requests in an endless loop, i.e. according to my logfile it floods the poor Telegram server with about 10 POST requests per second,

If this is done by all Telepot clients on the net, BotFather will not be amused!

**Comment by nickoala** [comment]
Someone raised this issues before, but did not raise my attention enough because he did not portray it as an DDoS attack!!! :scream: :scream: :scream: :scream: :scream: :scream: 

Ok, I get your point. Weighing the efforts required and the potential benefits .... I think I will fix it in a later version.

**Comment by cooox** [comment]
Problem seems to be the `relax` value in combination with a generic `except:` catch-all that catches an invalid JSON response from the Telegram API. 

I guess this could be solved by adding the following code block into [`/telepot/__init__.py`](https://github.com/nickoala/telepot/blob/master/telepot/__init__.py#L813).

```python
    offset = max([relay_to_collector(update) for update in result]) + 1

except exception.BadHTTPResponse as ex:
    traceback.print_exc()

    # Handle Telegram gateway problems with longer sleep time
    # to prevent a retry after 0.1 seconds (the default relax value)
    if ex.status == 502:
        time.sleep(60)

except:
    traceback.print_exc()

finally:
    time.sleep(relax)
```

**Comment by nickoala** [comment]
Fixed in 10.5. Thanks to @long-exposure for reporting. Thanks to @cooox for the solution.
